.github/workflows/build.yml

```diff
@@ -246,3 +245,50 @@ jobs:
           dev_tags: |
             hashicorppreview/${{ env.repo }}:${{ env.version }}
             hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}
+
+  upload-dev-docker:
```

Comment:
I'm probably missing some context here, but just curious about the use case for pushing the image directly from this workflow, vs relying on the `dev_tags` specified above? If dev_tags aren't satisfying your use case I wonder if we could make some changes to CRT to better support this project?

Reply:
We have acceptance tests we run pre-merge to test PRs (so for non-release branches, basically). Those tests spin up resources in AWS, so we need a public image that AWS can pull in order to run those tests.

`dev_tags` is great, but the CRT workflows only run for release branches, and they publish "stable" artifacts. Here, I publish an "unstable" image per-commit for all branches. This works well for us - we can test with any commit of any branch - and is quick since it doesn't wait for the sign/notarize steps.

I asked in `#team-rel-eng` about doing this with CRT. We could update the `.release/ci.hcl` in each PR to temporarily include that PR branch in the list of release branches, so that the CRT workflow can run. But, this requires manual steps on every PR, and puts unstable artifacts into the "stable" bucket, which is not ideal.